tiledb/CMakeLists.txt

```diff
@@ -533,23 +533,44 @@ endif()
 # Azure dependencies
 if (TILEDB_AZURE)
   message(STATUS "The TileDB library is compiled with Azure support.")
-  find_package(azure-storage-blobs-cpp CONFIG REQUIRED)
-  if (NOT TARGET LibLZMA)
-	  find_package(LibLZMA REQUIRED)
-  endif()
-  if (NOT TARGET LibXml2::LibXml2)
+
+  if (TILEDB_VCPKG)
+    find_package(azure-storage-blobs-cpp CONFIG REQUIRED)
+    if (NOT TARGET LibLZMA)
+      find_package(LibLZMA REQUIRED)
+    endif()
+    if (NOT TARGET LibXml2::LibXml2)
       find_package(libxml2 CONFIG REQUIRED)
       install_target_libs(LibXml2::LibXml2)
+    endif()
+    target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
+            INTERFACE
+            Azure::azure-storage-blobs
+            Azure::azure-storage-common
+            Azure::azure-core
+            LibXml2::LibXml2
+            LibLZMA::LibLZMA
+            )
+    install_all_target_libs("Azure::azure-storage-blobs;Azure::azure-storage-common;Azure::azure-core")
+  else()
+    find_package(AzureCore_EP REQUIRED)
+    find_package(AzureStorageCommon_EP REQUIRED)
+    find_package(AzureStorageBlobs_EP REQUIRED)
+    if (NOT TARGET LibXml2::LibXml2)
+      find_package(libxml2 REQUIRED)
+      install_target_libs(LibXml2::LibXml2)
+    endif()
+    target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
+            INTERFACE
+            Azure::azure-storage-common
+            Azure::azure-storage-blobs
+            Azure::azure-core
+            LibXml2::LibXml2)
+    if(MSVC)
```

Comment:
```suggestion
    if(WIN32)
```

Reply:
Well I specifically used MSVC because I'm guessing (but haven't tried yet) that with mingw we are not going to need/want to link this.